flink-runtime/src/main/java/org/apache/flink/runtime/jobmaster/slotpool/SlotPoolImpl.java

```diff
@@ -592,6 +589,31 @@ private PendingRequest findMatchingPendingRequest(final AllocatedSlot slot) {
 		return null;
 	}
 
+	private void maybeRemapOrphanedAllocation(
+			final AllocationID allocationIdOfRequest,
+			final AllocationID allocationIdOfSlot) {
+
+		final AllocationID orphanedAllocationId = allocationIdOfRequest.equals(allocationIdOfSlot)
+			? null : allocationIdOfRequest;
+
+		// if the request that initiated the allocation is still pending, it should take over the orphaned allocation
+		// of the fulfilled request so that it can fail fast if the remapped allocation fails
+		if (orphanedAllocationId != null) {
```

Comment:
```suggestion
		// if the request that initiated the allocation is still pending, it should take over the orphaned allocation
		// of the fulfilled request so that it can fail fast if the remapped allocation fails
		if (!allocationIdOfRequest.equals(allocationIdOfSlot)) {
		    final AllocationID orphanedAllocationId = allocationIdOfRequest;
```

Reply:
done.